src/services/organizeImports.ts

```diff
@@ -0,0 +1,291 @@
+
+            for (const importDeclaration of importGroup) {
+                if (importDeclaration.importClause === undefined) {
+                    // Only the first such import is interesting - the others are redundant.
+                    // Note: Unfortunately, we will lose trivia that was on this node.
+                    if (!seenImportWithoutClause) {
+                        coalescedImports.push(importDeclaration);
+                    }
+
+                    seenImportWithoutClause = true;
+                    continue;
+                }
+
+                const { name, namedBindings } = importDeclaration.importClause;
+
+                if (name) {
+                    defaultImports.push(name);
+                }
+
+                if (namedBindings) {
+                    if (isNamespaceImport(namedBindings)) {
+                        namespaceImports.push(namedBindings);
+                    }
+                    else {
+                        namedImports.push(namedBindings);
+                    }
+                }
+            }
+
+            // Normally, we don't combine default and namespace imports, but it would be silly to
+            // produce two import declarations in this special case.
+            if (defaultImports.length === 1 && namespaceImports.length === 1 && namedImports.length === 0) {
+                // Add the namespace import to the existing default ImportDeclaration.
+                const defaultImportClause = defaultImports[0].parent as ImportClause;
+                coalescedImports.push(
+                    updateImportDeclarationAndClause(defaultImportClause, defaultImportClause.name, namespaceImports[0]));
+
+                continue;
+            }
+
+            // For convenience, we cheat and do a little sorting during coalescing.
+            // Seems reasonable since we're restructuring so much anyway.
+            const sortedNamespaceImports = stableSort(namespaceImports, (n1, n2) => compareIdentifiers(n1.name, n2.name));
+
+            for (const namespaceImport of sortedNamespaceImports) {
+                // Drop the name, if any
+                coalescedImports.push(
+                    updateImportDeclarationAndClause(namespaceImport.parent, /*name*/ undefined, namespaceImport));
+            }
+
+            if (defaultImports.length === 0 && namedImports.length === 0) {
+                continue;
+            }
+
+            let newDefaultImport: Identifier = undefined;
+            const newImportSpecifiers: ImportSpecifier[] = [];
+            if (defaultImports.length === 1) {
+                newDefaultImport = defaultImports[0];
+            }
+            else {
+                for (const defaultImport of defaultImports) {
+                    newImportSpecifiers.push(
+                        createImportSpecifier(createIdentifier("default"), defaultImport));
+                }
+            }
+
+            for (const namedImport of namedImports) {

```

Comment:
`newImportSpecifiers.push(...flatMap(namedImports, n => n.elements));`

Reply:
Thanks, I figured there had to be a function that did that.  I was looking for "selectMany".